data-validation/check-input-datasets.js

```diff
@@ -0,0 +1,129 @@
+const fsPromises = require("fs").promises;
```

Comment:
how is this different from process-dataset/validate.js?  Should validate.js be removed in favor of this one? 

My understanding of how this works is not very good.  How are errors reported?  If it fails do you have to look in the gh actions console output?

Reply:
`process-dataset/validate.js` is just a command line script to validate one file at a time. I'm not sure it's being used anywhere, I could have adapted it to do what I wanted here, which is to both log the result but also collect the fact that an error occurred but still continue. 

So if any of the datasets fail the check, the action would fail and show as an `X` on the PR, but to know which dataset failed and why, you would have to click into the gh actions log